src/main/java/com/mongodb/spark/sql/connector/read/MongoScanBuilder.java

```diff
@@ -306,7 +305,9 @@ private Optional<BsonValue> getBsonValue(final String fieldName, final Object va
           localSchema = (StructType) localField.dataType();
         }
       }
-      return Optional.of(CONVERTER.toBsonValue(localDataType, value));
+      Function<Object, BsonValue> objectToBsonValue =
+          createObjectToBsonValue(localDataType, WriteConfig.ConvertJson.FALSE, false);
+      return Optional.of(objectToBsonValue.apply(value));
```

Comment:
Here the cost of the generating the filters is based on the number of filters applied to the `DataFrameReader`. In the case where non are applied there is no cost. This could be updated to cache the function on demand, but it would only be beneficial if the same fieldname had multiple filters applied to it.

As this is a single cost once per creation of the `DataFrameReader` its performance impact is much less than the purpose of this PR which changes the `DataFrameWriter` cost from: calculate for every `Row` to once per `DataFrameWriter`.

So until we have data saying this is a bottleneck I'd rather keep the code simpler.

Reply:
Ross, thank you for explaining.